### ISSUE TITLE
Better output when using with go  XUnit-style tests

### ISSUE THREAD

**Issue by smyrman** [onsi/gomega#255]
It would be nice if the gomega matcher output could be more short / useful for XUnit tests in terms of printing the correct file/line numbers for the error, and nothing more.

My use-case is to use Gomega with a non-Ginko BDD-style framework called [spec](https://github.com/sclevine/spec), which rely on the built-in Go 7+ sub-tests for execution.

E.g. with the following example code (one test using Gomega, another one not using Gomega):
```go
package main_test

import (
	"testing"

	"github.com/onsi/gomega"
	"github.com/sclevine/spec"
)

func TestExample(t *testing.T) {
	spec.Run(t, "Example", func(t *testing.T, when spec.G, it spec.S) {
		var result *struct{ i int }

		it("should not be nil", func() {
			if result == nil {
				t.Error("result is nil")
			}
		})
	})
}

func TestExampleGomega(t *testing.T) {
	spec.Run(t, "Example", func(t *testing.T, when spec.G, it spec.S) {
		var result *struct{ i int }
		gomega.RegisterTestingT(t)
		it("should not be nil", func() {
			gomega.Expect(result).ToNot(gomega.BeNil())
		})
	})
}
```

I wold expect similar output from both tests. e.g.:
```
% go test -v
=== RUN   TestExample
=== RUN   TestExample/Example
=== RUN   TestExample/Example/should_not_be_nil
--- FAIL: TestExample (0.00s)
    --- FAIL: TestExample/Example (0.00s)
        --- FAIL: TestExample/Example/should_not_be_nil (0.00s)
        	my_test.go:16: result is nil
=== RUN   TestExampleGomega
=== RUN   TestExampleGomega/Example
=== RUN   TestExampleGomega/Example/should_not_be_nil
--- FAIL: TestExampleGomega (0.00s)
    --- FAIL: TestExampleGomega/Example (0.00s)
        --- FAIL: TestExampleGomega/Example/should_not_be_nil (0.00s)
        	my_test.go:16:
        		Expected
        		    <*struct { i int } | 0x0>: nil
        		not to be nil
FAIL
exit status 1
FAIL	github.com/searis/play/gomegaoutput	0.008s
```

Instead I get the filename / line number of a file internal to Gomega, and a trace-back that adds little value for my test:
```
% go test -v
=== RUN   TestExample
=== RUN   TestExample/Example
=== RUN   TestExample/Example/should_not_be_nil
--- FAIL: TestExample (0.00s)
    --- FAIL: TestExample/Example (0.00s)
        --- FAIL: TestExample/Example/should_not_be_nil (0.00s)
        	my_test.go:16: result is nil
=== RUN   TestExampleGomega
=== RUN   TestExampleGomega/Example
=== RUN   TestExampleGomega/Example/should_not_be_nil
--- FAIL: TestExampleGomega (0.00s)
    --- FAIL: TestExampleGomega/Example (0.00s)
        --- FAIL: TestExampleGomega/Example/should_not_be_nil (0.00s)
        	testing_t_support.go:22: 
        			/Users/smyrman/.local/src/github.com/searis/play/gomegaoutput/vendor/github.com/onsi/gomega/internal/assertion/assertion.go:69 +0x1ef
        		github.com/searis/play/gomegaoutput/vendor/github.com/onsi/gomega/internal/assertion.(*Assertion).ToNot(0xc42004eac0, 0x124fa40, 0x1321830, 0x0, 0x0, 0x0, 0xc42004eac0)
        			/Users/smyrman/.local/src/github.com/searis/play/gomegaoutput/vendor/github.com/onsi/gomega/internal/assertion/assertion.go:39 +0xae
        		github.com/searis/play/gomegaoutput_test.TestExampleGomega.func1.1()
        			/Users/smyrman/.local/src/github.com/searis/play/gomegaoutput/my_test.go:27 +0xbe
        		github.com/searis/play/gomegaoutput/vendor/github.com/sclevine/spec.run(0xc420042ec8, 0x1, 0x1)
        			/Users/smyrman/.local/src/github.com/searis/play/gomegaoutput/vendor/github.com/sclevine/spec/spec.go:182 +0x37
        		github.com/searis/play/gomegaoutput/vendor/github.com/sclevine/spec.Run.func3(0xc4200f85a0, 0xc420044760, 0x1, 0x1, 0xc4200124e8, 0x1, 0x1, 0x5a328010, 0x1, 0x1, ...)
        			/Users/smyrman/.local/src/github.com/searis/play/gomegaoutput/vendor/github.com/sclevine/spec/spec.go:176 +0x3af
        		github.com/searis/play/gomegaoutput/vendor/github.com/sclevine/spec.node.run.func1(0xc4200f85a0)
        			/Users/smyrman/.local/src/github.com/searis/play/gomegaoutput/vendor/github.com/sclevine/spec/parser.go:132 +0x54
        		testing.tRunner(0xc4200f85a0, 0xc42000a5a0)
        			/usr/local/Cellar/go/1.9.1/libexec/src/testing/testing.go:746 +0xd0
        		created by testing.(*T).Run
        			/usr/local/Cellar/go/1.9.1/libexec/src/testing/testing.go:789 +0x2de
        		
        		Expected
        		    <*struct { i int } | 0x0>: nil
        		not to be nil
FAIL
exit status 1
FAIL	github.com/searis/play/gomegaoutput	0.008s
```

PS! The following recent addition to the Go testing library may be of some use in prettifying this:
- https://golang.org/pkg/testing/#T.Helper

**Comment by onsi** [comment]
I think I've got line of sight on a fix for this - will involve using T.Helper.

**Comment by onsi** [comment]
ok - so I've pushed something out to a branch called `stacktrace`

tl;dr: I now string t.Helper along and call it in every function that needs to be popped off the stack.  I no longer render my own stacktrace and rely solely on `go test`'s stack trace - so this should feel much more "native"

Please give it a try and let me know what you think @smyrman

**Comment by smyrman** [comment]
I will give it a try and see how it works now.

**Comment by onsi** [comment]
thanks - note you'll have to be on Go 1.9 - it's probably a terrible experience on older versions of Go.  I can try to fix that..

**Comment by onsi** [comment]
latest code should produce more useful output for older versions of Go too, now.

**Comment by AlTavares** [comment]
Hey, is this going to be merged anytime soon?
Thanks!

**Comment by blgm** [comment]
That's a very good question.  I've raised PR https://github.com/onsi/gomega/pull/297 to merge the changes.  That's no guarantee that they will be merged, but it at least keeps it on the radar.  I fear the branch has been forgotten about.
